File: bindings/rust/libgpiod/examples/get_chip_info.rs
// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
// SPDX-FileCopyrightText: 2023 Kent Gibson <warthog618@gmail.com>
//
// Minimal example of reading the info for a chip.

use libgpiod::{self, chip::Chip, Result};

fn main() -> Result<()> {
    // Example configuration - customize to suit your situation
    let chip_path = "/dev/gpiochip0";

    let chip = Chip::open(&chip_path)?;
    let info = chip.info()?;
    println!(
        "{} [{}] ({} lines)",
        info.name()?,
        info.label()?,
        info.num_lines(),
    );

    Ok(())
}

File: bindings/cxx/examples/get_chip_info.cpp
// SPDX-License-Identifier: GPL-2.0-or-later
// SPDX-FileCopyrightText: 2023 Kent Gibson <warthog618@gmail.com>

/* Minimal example of reading the info for a chip. */

#include <cstdlib>
#include <filesystem>
#include <gpiod.hpp>
#include <iostream>

namespace {

/* Example configuration - customize to suit your situation */
const ::std::filesystem::path chip_path("/dev/gpiochip0");

} /* namespace */

int main()
{
	::gpiod::chip chip(chip_path);
	auto info = chip.get_info();

	::std::cout << info.name() << " [" << info.label() << "] ("
		    << info.num_lines() << " lines)" << ::std::endl;

	return EXIT_SUCCESS;
}
